### DISCUSSION TITLE
Issue on MacOS - Not logged in error in log.txt

### DISCUSSION THREAD

**Discussion by iamrsoze** [anilabhadatta/educative.io_scraper#33] [Q&A] [answered]
Unable to scrap the content. 
Chromedriver version - 110.0.5481.77

**Comment by anilabhadatta** [comment]
@iamrsoze I need more details, and a screenshot of the terminal where the error occurred.
The chrome driver version should be 98 since the binary provided is version 98 and it should be cloned automatically. 
Are you using an M1 chip or Intel?
Send me the screenshot of these pages by opening the downloaded chrome binary manually or via script(login method-3):
chrome://version/
chrome://settings/help

**Comment by iamrsoze** [comment]
It was the incorrect version of Chrome browser that was causing the issue. Works smoothly on Chrome 98.
However, my chrome starts auto updating after a while and the scrapper stops due to version mismatch error.

Any leads on how can I stop that?
PS - I looked on SO for solutions but none of them helped.

**Reply by anilabhadatta** [reply]
@iamrsoze if you are using the chrome binary from my github repo: https://github.com/anilabhadatta/educative.io_scraper/releases/download/Educative-Scraper-v6.8/Chrome-bin-osx.zip
It shouldn't update since I removed the updating executable. If it is still updating then please reply, I will look into the updating part.

**Comment by iamrsoze** [comment]
The binary from the github repo isn't working for me (attached screenshot).
I'm using Intel Mac Pro, MacOS Ventura 13.1

<img width="985" alt="Screenshot 2023-03-03 at 11 15 24 PM" src="https://user-images.githubusercontent.com/126609058/222791600-b243e41e-c101-4329-917b-acb82a5ba401.png">

**Reply by anilabhadatta** [reply]
@iamrsoze go to settings -> securty -> general -> you will find chrome asking for permission to open

**Comment by anilabhadatta** [comment]
@iamrsoze any updates?

**Comment by iamrsoze** [comment]
I know what you're saying to do a "Allow Anyway" in Settings -> Privacy & Security.
But I am not getting any prompt to approve "chrome asking for permission". This error is different. 
It is showing the file as damaged in first place.

PS - I don't have any other chrome installed on Mac. I am just copying this in Chrome-bin folder and trying to open via command line.

**Comment by anilabhadatta** [comment, marked as answer]
@iamrsoze 
Follow this tutorial
https://user-images.githubusercontent.com/48487849/222886869-4f815acf-cde0-4aff-b978-9c66c598ef9c.mp4

Step 1: Double click on Chromium app to open, If it shows damaged then check for Security and Privacy -> General Settings to allow access to chromium.
Step 2: If it still shows then open a terminal inside mac folder and run the following command : xattr -cr Chromium.app
Now it should fix the issue and chromium should open up if it doesn't recheck step 1.

**Reply by iamrsoze** [reply]
xattr did it for me. Thanks!

**Reply by anilabhadatta** [reply]
@iamrsoze Welcome

**Reply by iamrsoze** [reply]
UPDATE : The chrome is still updating after some time. Once I have scrapped a course completely and I try to run for a new course, I get following -


```
Found Issue, Going Next Course Message: session not created: This version of ChromeDriver only supports Chrome version 98
Current browser version is 110.0.5481.177 with binary path /Users/darsh/Documents/educative.io_scraper/Chrome-bin/mac/Chromium.app/Contents/MacOS/Google Chrome
```

**Reply by anilabhadatta** [reply]
@iamrsoze for now i can suggest you to update the chromedriver to version 110

**Reply by anilabhadatta** [reply]
@iamrsoze https://github.com/anilabhadatta/enable-disable-chrome-updates Refer to this repo to enable/disable updates . Download and extract fresh Chrome-bin (updated just now) from my latest releases and the chromedriver for v98 from the repo or official website.

**Comment by anilabhadatta** [comment]
@iamrsoze new update was released few days ago, reclone to a new location and test it in your system
